src/swap/sideshift.js

```diff
@@ -0,0 +1,301 @@
+  opts: EdgeCorePluginOptions
+): EdgeSwapPlugin {
+  const { io, initOptions } = opts
+  const { fetch = io.fetch } = io
+  const baseUrl = 'https://sideshift.ai/api/'
+
+  async function get(path: string): Promise<any> {
+    const url = `${baseUrl}${path}`
+    const reply = await fetch(url)
+    return reply
+  }
+
+  async function post(path, body): Promise<any> {
+    const url = `${baseUrl}${path}`
+    const reply = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(body)
+    })
+    return checkReplyForError(url, reply)
+  }
+
+  const out: EdgeSwapPlugin = {
+    swapInfo,
+    async fetchSwapQuote(request: EdgeSwapRequest): Promise<EdgeSwapQuote> {
+      const permission: Permission = await get('permissions')
+
+      if (!permission.createOrder) {
+        throw new SwapPermissionError(swapInfo, 'geoRestriction')
+      }
+
+      const [depositAddress, settleAddress] = await Promise.all([
+        getAddress(request.fromWallet, request.fromCurrencyCode),
+        getAddress(request.toWallet, request.toCurrencyCode)
+      ])
+      const quoteAmount =
+        request.quoteFor === 'from'
+          ? await request.fromWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.fromCurrencyCode
+            )
+          : await request.toWallet.nativeToDenomination(
+              request.nativeAmount,
+              request.toCurrencyCode
+            )
+
+      let safeFromCurrencyCode = request.fromCurrencyCode.toLowerCase()
+      let safeToCurrencyCode = request.toCurrencyCode.toLowerCase()
+      if (CURRENCY_CODE_TRANSCRIPTION[request.fromCurrencyCode]) {
+        safeFromCurrencyCode =
+          CURRENCY_CODE_TRANSCRIPTION[request.fromCurrencyCode]
+      }
+      if (CURRENCY_CODE_TRANSCRIPTION[request.toCurrencyCode]) {
+        safeToCurrencyCode = CURRENCY_CODE_TRANSCRIPTION[request.toCurrencyCode]
+      }
+
+      const ratePath =
+        request.quoteFor === 'from'
+          ? `pairs/${safeFromCurrencyCode}/${safeToCurrencyCode}`
+          : `pairs/${safeToCurrencyCode}/${safeFromCurrencyCode}`
+
+      const rate: Rate = await get(ratePath)
+
+      // TODO: not sure if this part is needed. If /quotes request throws an error if settleAmount is below min/ above max, it can be removed
+      await checkRateForError(rate, request)
+
+      const fixedRateQuoteParams: FixedQuoteRequestParams = {
+        depositMethod: safeFromCurrencyCode,
+        settlMethod: safeToCurrencyCode,
+        depositAmount: quoteAmount
+      }
+
+      const fixedRateQuote: FixedQuote = await post(
+        'quotes',
+        fixedRateQuoteParams
+      )
+
+      const orderRequestParams: OrderRequestParams = {
+        type: 'fixed',
+        quoteId: fixedRateQuote.id,
+        affiliateId: initOptions.affiliateId, // TODO: hardcode it in the .env.json
+        sessionSecret: 'this can be empty right?',
```

Comment:
`sessionSecret` at `createOrder` is only needed to prove ownership of the order, in case the shift fails and the user/wallet wants to claim the refund (by providing a refund address).

As far as I know, we won't have this flow in the Edge wallet. (@abrkn?)